js_modules/dagster-ui/packages/ui-core/src/assets/catalog/AssetCatalogTableV2.tsx

```diff
@@ -158,24 +132,23 @@ export const AssetCatalogTableV2 = React.memo(() => {
   });
 
   const displayKeys = useMemo(() => {
-    return Object.values(groupedByStatus).flatMap((assets) =>
-      assets.map((asset) => JSON.stringify(asset.key.path)),
+    return Object.values(groupedAndSorted).flatMap((group) =>
+      group.assets.map((asset) => tokenForAssetKey(asset.key)),
     );
-  }, [groupedByStatus]);
+  }, [groupedAndSorted]);
 
   const [{checkedIds: checkedDisplayKeys}, {onToggleFactory}] = useSelectionReducer(displayKeys);
 
   const onToggleGroup = useCallback(
-    (status: AssetHealthStatusString) => {
+    (group: string) => {
       return (checked: boolean) => {
-        const assetsInGroup = groupedByStatus[status];
-        assetsInGroup.forEach((asset) => {
-          const toggle = onToggleFactory(JSON.stringify(asset.key.path));
+        groupedAndSorted[group]?.assets.forEach((asset) => {
+          const toggle = onToggleFactory(tokenForAssetKey(asset.key));
```

Comment:
Don't imagine perf is really a concern here, but this does dispatch into useReducer a lot of times in a row. I wonder if it'd be worth making `onToggleAll` take an array of ids and running this as a single setState instead.